src/ESLint.Formatter/sarif.js

```diff
@@ -54,13 +54,20 @@ module.exports = function (results, data) {
 
     const sarifFiles = {};
     const sarifArtifactIndices = {};
-    var nextArtifactIndex = 0;
+    let nextArtifactIndex = 0;
     const sarifRules = {};
     const sarifRuleIndices = {};
-    var nextRuleIndex = 0;
+    let nextRuleIndex = 0;
     const sarifResults = [];
     const embedFileContents = process.env.SARIF_ESLINT_EMBED === "true";
 
+    // Emit a tool execution notification with this id if ESLint emits a message with
+    // no ruleId (which indicates an internal error in ESLint).
+    const internalErrorId = "ESL9999"
```

Comment:
>ESL9999 [](start = 29, length = 7)

Suggest using ESL0999 for this. A poorly adopted convention with MS compilers is for the 1000 - 1999 range to be preserved for core compiler/language errors and everything beyond is for additional analysis. IDs below 1000 are reserved for internal engine issues.

Reply:
Fixed.

---
In reply to: [385957944](https://github.com/microsoft/sarif-sdk/pull/1793#discussion_r385957944) [](ancestors = 385957944)